apps/api/src/api.controller.ts

```diff
@@ -17,4 +19,76 @@ export class ApiController {
       status: HttpStatus.OK,
     };
   }
+
+  @Put('asset/:userDsnpId/upload')
+  @UseInterceptors(FilesInterceptor('files'))
+  @HttpCode(202)
+  @ApiConsumes('multipart/form-data')
+  @ApiBody({
+    description: 'Asset files',
+    type: FilesUploadDto,
+  })
+  // eslint-disable-next-line no-undef
+  async assetUpload(@Param('userDsnpId') userDsnpId: string, @UploadedFiles() files: Array<Express.Multer.File>): Promise<UploadResponseDto> {
+    this.logger.log(`broadcast ${userDsnpId}`);
+    this.logger.log(files.length);
+    // this.logger.log(JSON.stringify(files));
+    return {
+      assetIds: files.map((_) => uuidv4()),
+    };
+  }
+
+  @Post('content/:userDsnpId/broadcast')
+  @HttpCode(202)
+  async broadcast(@Param('userDsnpId') userDsnpId: string, @Body() broadcastDto: BroadcastDto): Promise<AnnouncementResponseDto> {
```

Comment:
Not this PR, but I wonder how we will reuse the announcement structure while still waiting to upload the files.

I guess the upload endpoint could return the ipfs.io CID url (as it knows the upload parameters, we can generate the correct CID) and so while not uploaded yet, the incoming `broadcastDto` That way it is just matching the CID from the ifps url to trigger the release to IPFS from the upload queue.

Reply:
I'm not sure if I follow the reuse scenario that you are mentioning. Would you please elaborate on that?

Reply:
Mostly how the `assetIds` strings are used. If those `assetIds` were urls of the form `https://ipfs.io/ipfs/[CID]`, then we could just insert them into the Broadcast/ReplyDto from the calling service. Then in here, it would just be a matter of parsing, validating, and triggering off the actual ipfs pins of those files inside the BroadcastDto.

Reply:
I think there are some confusion about how we are imagining this might work. I'll create a diagram before start implementing that part to make sure we are on the same page.